src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -1322,23 +1283,10 @@ impl<'a, 'tcx> Visitor<'tcx> for Checker<'a, 'tcx> {
                     match self.tcx.fn_sig(def_id).abi() {
                         Abi::RustIntrinsic |
                         Abi::PlatformIntrinsic => {
-                            assert!(!self.tcx.is_const_fn(def_id));
+                            assert!(self.tcx.is_const_fn(def_id),
```

Comment:
Oh... this is not for const fn, but for normal functions. So you can remove the assert since it's now useless and you can also remove the `RustIntrinsic` pattern above, as we only seem to care about SIMD_shuffle which is a PlatformIntrinsic